### ISSUE TITLE
Getting agents configuration crash for some calls

### ISSUE THREAD

**Issue by chemamartinez** [wazuh/wazuh-api#154]
Found on the branch `dev-query-config` (it is not a existing bug on the last released version).

It has been found this error when getting the configuration as follows:

```
root@ubuntu:~# curl -u foo:bar -k -X GET "http://127.0.0.1:55000/agents/001/config/agent/global?pretty"
{
   "error": 1000,
   "message": "Wazuh-Python Internal Error: list index out of range"
}
```

And the full trace:

```
root@ubuntu:~# echo '{"function":"/agents/:agent_id/config/:component/:configuration","arguments":{"agent_id":"001","component":"agent","configuration":"client"},"ossec_path":"/var/ossec/"}' | /var/ossec/api/models/wazuh-api.py --debug
{"message": "Wazuh-Python Internal Error: list index out of range", "error": 1000}
Traceback (most recent call last):
  File "/var/ossec/api/models/wazuh-api.py", line 290, in <module>
    data = functions[request['function']](**request['arguments'])
  File "/var/ossec/framework/wazuh/agent.py", line 2813, in get_config
    return Agent(agent_id).getconfig(component=component, configuration=configuration)
  File "/var/ossec/framework/wazuh/agent.py", line 2791, in getconfig
    rec_msg = data[1]
IndexError: list index out of range
```

**Comment by druizz90** [comment]
I made that `IndexError: list index out of range` was catched on a try/catch block and now shows this message in branch https://github.com/wazuh/wazuh/tree/fix-query-config:

```bash
[root@localhost ~]# curl -u foo:bar -k -X GET "http://127.0.0.1:55000/agents/001/config/agent/global?pretty"
{
   "error": 1014,
   "message": "Error communicating with socket: Data could not be received"
}
```
Furthermore, agent version is checked because getting agents configuration is only available since 3.7 version.

**Comment by TJOSERAFAEL** [comment]
The fix is perfect because you may connect to a socket that doesn't sends the message length at the beginning .
